Error: unsupported attribute "content_base64"

<!--
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

If your issue relates to a specific Terraform provider, please open it in the provider's own repository. The index of providers is at https://github.com/terraform-providers .
-->

### Summary
At one point I tried the content_base64 option for local_file on one of my config files. Then I reverted this change back to content. After doing this, each local_file resource that I have produces a `unsupported attribute "content_base64"` error. I have used `grep base64 *.tf` and cannot find any reference to this setting throughout my terraform configuration. I've provided a sample local_file config below that is failing according to the DEBUG output, also provided below. The only workaround is to delete all local files and also remove them from the terraform state, but the error continues to crop up once terraform has created the files anew. Nothing is wrong with the files otherwise. Very strange and currently I'm considering wiping our state file and simply starting with a new apply since I don't know what else could be causing this.

### Terraform Version
<!---
Run `terraform -v` to show the version, and paste the result between the ``` marks below.

If you are not running the latest version of Terraform, please try upgrading because your issue may have already been fixed.
-->

```
→ terraform -v
Terraform v0.12.3
+ provider.aws v2.15.0
+ provider.local v1.2.2
+ provider.null v2.1.2
+ provider.random v2.1.2
+ provider.template v2.1.2
```

### Terraform Configuration Files
<!--
Paste the relevant parts of your Terraform configuration between the ``` marks below.

For large Terraform configs, please use a service like Dropbox and share a link to the ZIP file. For security, you can also encrypt the files using our GPG public key.
-->

```
resource "local_file" "pgpass" {
  content = "${aws_db_instance.db.endpoint}:db-example:${aws_db_instance.db.username}:${var.db-password}"
  filename = "../pgpass-${terraform.workspace}"
}

```

### Debug Output
<!--
Full debug output can be obtained by running Terraform with the environment variable `TF_LOG=trace`. Please create a GitHub Gist containing the debug output. Please do _not_ paste the debug output in the issue, since debug output is long.

Debug output may contain sensitive information. Please review it before posting publicly, and if you are concerned feel free to encrypt the files using the HashiCorp security public key.
-->
```
2019/07/09 09:54:51 [DEBUG] ReferenceTransformer: "local_file.pgpass" references: []
2019/07/09 09:54:51 [TRACE] Completed graph transform *terraform.ReferenceTransformer (no changes)
2019/07/09 09:54:51 [TRACE] Executing graph transform *terraform.RootTransformer
2019/07/09 09:54:51 [TRACE] Completed graph transform *terraform.RootTransformer (no changes)
2019/07/09 09:54:51 [ERROR] <root>: eval: *terraform.EvalReadState, err: unsupported attribute "content_base64"
2019/07/09 09:54:51 [ERROR] <root>: eval: *terraform.EvalSequence, err: unsupported attribute "content_base64"
2019/07/09 09:54:51 [TRACE] [walkRefresh] Exiting eval tree: local_file.pgpass
2019/07/09 09:54:51 [TRACE] vertex "local_file.pgpass": visit complete
2019/07/09 09:54:51 [TRACE] vertex "local_file.pgpass": dynamic subgraph encountered errors
```

### Crash Output
<!--
If the console output indicates that Terraform crashed, please share a link to a GitHub Gist containing the output of the `crash.log` file.
-->
`Error: unsupported attribute "content_base64"`

### Expected Behavior
<!--
What should have happened?
-->
Should create the file without content_base64 and without error.

### Actual Behavior
<!--
What actually happened?
-->
Consistently getting content_base64 error and failing to apply terraform configuration.

### Steps to Reproduce
<!--
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform apply`
-->
1. Run `terraform apply` and allow the deployment to complete successfully.
2. Make any normal config change to a local file.
3. Run `terraform apply`
4. Get content_base64 error.

I am unable to reproduce the problem outside of the large project that I'm in. I've attempted to set a single local_file to content_base64 and back to content, but it does not have the same problem.

### Additional Context
<!--
Are there anything atypical about your situation that we should know? For example: is Terraform running in a wrapper script or in a CI system? Are you passing any unusual command line options or environment variables to opt-in to non-default behavior?
-->
I believe I am in a bad state and I'm considering wiping my state file. I attempted to try content_base64 on a single local_file configuration, but now each time I run terraform apply and the files exist, I get the "content_base64" error for each file that is being created locally.

### Workaround
Clear all local files from both terraform and locally and terraform apply.

### References
<!--
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:

- #6017

-->

--- a/command/show_test.go
+++ b/command/show_test.go
@@ -5,6 +5,7 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"strings"
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
@@ -153,6 +154,8 @@ func TestShow_json_output(t *testing.T) {
 			defer os.RemoveAll(td)
 			defer testChdir(t, td)()
 
+			expectError := strings.Contains(entry.Name(), "error")
+
 			p := showFixtureProvider()
 			ui := new(cli.MockUi)
 			m := Meta{
@@ -171,6 +174,10 @@ func TestShow_json_output(t *testing.T) {
 				},
 			}
 			if code := ic.Run([]string{}); code != 0 {
+				if expectError {
+					// this should error, but not panic.
+					return
+				}
 				t.Fatalf("init failed\n%s", ui.ErrorWriter)
 			}
 

--- a/command/testdata/show-json/nested-module-error/main.tf
+++ b/command/testdata/show-json/nested-module-error/main.tf
@@ -1,0 +1,3 @@
+module "my_module" {
+  source = "./modules"
+}

--- a/command/testdata/show-json/nested-module-error/modules/main.tf
+++ b/command/testdata/show-json/nested-module-error/modules/main.tf
@@ -1,0 +1,3 @@
+module "more" {
+  source = "./more-modules"
+}

--- a/command/testdata/show-json/nested-module-error/modules/more-modules/main.tf
+++ b/command/testdata/show-json/nested-module-error/modules/more-modules/main.tf
@@ -1,0 +1,4 @@
+variable "misspelled" {
+  default     = "ehllo"
+  descriptoni = "I am a misspelled attribute"
+}

--- a/command/testdata/show-json/nested-modules/modules/more-modules/main.tf
+++ b/command/testdata/show-json/nested-modules/modules/more-modules/main.tf
@@ -1,4 +1,4 @@
-variable "misspelled" {
-  default     = "ehllo"
-  descriptoni = "I am a misspelled attribute"
+variable "ok" {
+  default     = "something"
+  description = "description"
 }

--- a/command/testdata/show-json/nested-modules/output.json
+++ b/command/testdata/show-json/nested-modules/output.json
@@ -1,23 +1,31 @@
 {
-    "format_version": "0.1",
-    "terraform_version": "0.12.1-dev",
-    "planned_values": {
-        "root_module": {}
-    },
-    "configuration": {
-        "root_module": {
+  "format_version": "0.1",
+  "terraform_version": "0.12.1-dev",
+  "planned_values": {
+    "root_module": {}
+  },
+  "configuration": {
+    "root_module": {
+      "module_calls": {
+        "my_module": {
+          "source": "./modules",
+          "module": {
             "module_calls": {
-                "my_module": {
-                    "source": "./modules",
-                    "module": {
-                        "module_calls": {
-                            "more": {
-                                "module": {}
-                            }
-                        }
+              "more": {
+                "module": {
+                  "variables": {
+                    "ok": {
+                      "default": "something",
+                      "description": "description"
                     }
-                }
+                  }
+                },
+                "source": "./more-modules"
+              }
             }
+          }
         }
+      }
     }
+  }
 }

--- a/configs/config_build.go
+++ b/configs/config_build.go
@@ -76,6 +76,7 @@ func buildChildModules(parent *Config, walker ModuleWalker) (map[string]*Config,
 		}
 
 		child.Children, modDiags = buildChildModules(child, walker)
+		diags = append(diags, modDiags...)
 
 		ret[call.Name] = child
 	}

--- a/configs/config_build_test.go
+++ b/configs/config_build_test.go
@@ -69,3 +69,48 @@ func TestBuildConfig(t *testing.T) {
 		t.Fatalf("child_a.child_c is same object as child_b.child_c; should not be")
 	}
 }
+
+func TestBuildConfigDiags(t *testing.T) {
+	parser := NewParser(nil)
+	mod, diags := parser.LoadConfigDir("testdata/nested-errors")
+	assertNoDiagnostics(t, diags)
+	if mod == nil {
+		t.Fatal("got nil root module; want non-nil")
+	}
+
+	versionI := 0
+	cfg, diags := BuildConfig(mod, ModuleWalkerFunc(
+		func(req *ModuleRequest) (*Module, *version.Version, hcl.Diagnostics) {
+			// For the sake of this test we're going to just treat our
+			// SourceAddr as a path relative to our fixture directory.
+			// A "real" implementation of ModuleWalker should accept the
+			// various different source address syntaxes Terraform supports.
+			sourcePath := filepath.Join("testdata/nested-errors", req.SourceAddr)
+
+			mod, diags := parser.LoadConfigDir(sourcePath)
+			version, _ := version.NewVersion(fmt.Sprintf("1.0.%d", versionI))
+			versionI++
+			return mod, version, diags
+		},
+	))
+
+	wantDiag := `testdata/nested-errors/child_c/child_c.tf:5,1-8: ` +
+		`Unsupported block type; Blocks of type "invalid" are not expected here.`
+	assertExactDiagnostics(t, diags, []string{wantDiag})
+
+	// we should still have module structure loaded
+	var got []string
+	cfg.DeepEach(func(c *Config) {
+		got = append(got, fmt.Sprintf("%s %s", strings.Join(c.Path, "."), c.Version))
+	})
+	sort.Strings(got)
+	want := []string{
+		" <nil>",
+		"child_a 1.0.0",
+		"child_a.child_c 1.0.1",
+	}
+
+	if !reflect.DeepEqual(got, want) {
+		t.Fatalf("wrong result\ngot: %swant: %s", spew.Sdump(got), spew.Sdump(want))
+	}
+}

--- a/configs/testdata/nested-errors/child_a/child_a.tf
+++ b/configs/testdata/nested-errors/child_a/child_a.tf
@@ -1,0 +1,4 @@
+
+module "child_c" {
+  source = "child_c"
+}

--- a/configs/testdata/nested-errors/child_c/child_c.tf
+++ b/configs/testdata/nested-errors/child_c/child_c.tf
@@ -1,0 +1,6 @@
+output "hello" {
+  value = "hello"
+}
+
+invalid "block" "type " {
+}

--- a/configs/testdata/nested-errors/root.tf
+++ b/configs/testdata/nested-errors/root.tf
@@ -1,0 +1,3 @@
+module "child_a" {
+  source = "child_a"
+}

